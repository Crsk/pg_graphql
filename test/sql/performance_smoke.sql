begin;

    create table author(
        id int primary key,
        a int,
        aa int,
        aaa int,
        aaaa int,
        aaaaa int,
        aaaaaa int,
        aaaaaaa int,
        aaaaaaaa int
    );

    create table book(
        id int primary key,
        author_id int references author(id),
        a int,
        aa int,
        aaa int,
        aaaa int,
        aaaaa int,
        aaaaaa int,
        aaaaaaa int,
        aaaaaaaa int
    );


    select jsonb_pretty(
        graphql.resolve($$
    {
      bookCollection {
        edges {
          node {
            id
            author {
              id
              bookCollection {
                edges {
                  node {
                    id
                    author {
                      id
                      bookCollection {
                        edges {
                          node {
                            id
                            author {
                              id
                              bookCollection {
                                edges {
                                  node {
                                    id
                                    author {
                                      id
                                      bookCollection {
                                        edges {
                                          node {
                                            id
                                            author {
                                              id
                                              bookCollection {
                                                edges {
                                                  node {
                                                    id
                                                    author {
                                                      id
                                                      bookCollection {
                                                        edges {
                                                          node {
                                                            id
                                                            author {
                                                              id
                                                              bookCollection {
                                                                edges {
                                                                  node {
                                                                    id
                                                                    author {
                                                                      id
                                                                      bookCollection {
                                                                        edges {
                                                                          node {
                                                                            id
                                                                          }
                                                                        }
                                                                      }
                                                                    }
                                                                  }
                                                                }
                                                              }
                                                            }
                                                          }
                                                        }
                                                      }
                                                    }
                                                  }
                                                }
                                              }
                                            }
                                          }
                                        }
                                      }
                                    }
                                  }
                                }
                              }
                            }
                          }
                        }
                      }
                    }
                  }
                }
              }
            }
          }
        }
      }
    }
        $$)
    );


rollback;
