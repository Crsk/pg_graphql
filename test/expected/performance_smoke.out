begin;
    create table author(
        id int primary key
    );
    create table book(
        id int primary key,
        author_id int references author(id)
    );
    select jsonb_pretty(
        graphql.resolve($$
    {
      bookCollection {
        edges {
          node {
            id
            author {
              id
              bookCollection {
                edges {
                  node {
                    id
                    author {
                      id
                      bookCollection {
                        edges {
                          node {
                            id
                            author {
                              id
                              bookCollection {
                                edges {
                                  node {
                                    id
                                    author {
                                      id
                                      bookCollection {
                                        edges {
                                          node {
                                            id
                                            author {
                                              id
                                              bookCollection {
                                                edges {
                                                  node {
                                                    id
                                                    author {
                                                      id
                                                      bookCollection {
                                                        edges {
                                                          node {
                                                            id
                                                            author {
                                                              id
                                                              bookCollection {
                                                                edges {
                                                                  node {
                                                                    id
                                                                    author {
                                                                      id
                                                                      bookCollection {
                                                                        edges {
                                                                          node {
                                                                            id
                                                                          }
                                                                        }
                                                                      }
                                                                    }
                                                                  }
                                                                }
                                                              }
                                                            }
                                                          }
                                                        }
                                                      }
                                                    }
                                                  }
                                                }
                                              }
                                            }
                                          }
                                        }
                                      }
                                    }
                                  }
                                }
                              }
                            }
                          }
                        }
                      }
                    }
                  }
                }
              }
            }
          }
        }
      }
    }
        $$)
    );
        jsonb_pretty         
-----------------------------
 {                          +
     "data": {              +
         "bookCollection": {+
             "edges": [     +
             ]              +
         }                  +
     }                      +
 }
(1 row)

rollback;
